pkg/handlers/generic/mutation/imageregistries/credentials/inject_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package credentials
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_needImageRegistryCredentials(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name    string
+		configs []providerConfig
+		need    bool
+		wantErr error
+	}{
+		{
+			name: "ECR credentials",
+			configs: []providerConfig{
+				{URL: "https://123456789.dkr.ecr.us-east-1.amazonaws.com"},
+			},
+			need: true,
```

Comment:
Why is this true? Credentials aren't required for ECR.

Reply:
Renamed the function to `needImageRegistryCredentialsConfiguration` since it was misleading that credentials were needed.